### PR TITLE
Eip 48 - concurrency configuration parameters

### PR DESCRIPTION
**Pull request by carlospzurita**
Fixing for https://github.com/etf-validator/governance/issues/48

We added in the configuration files a couple of parameters to control the queue and maximum number of threads. This sets to a default of the number of processors for maximum thread number, and 3 times this number for the queue size

### CODE REVIEW

**Comment by carlospzurita** [comment]
The remaining conflicts  are caused by the addition of our own code for the solution. The original code from the `next` branch is kept by this merge.

**Comment by carlospzurita** [comment]
As far as we see, it has been planned for milestone 2.1.0.
Just to clarify, from our point of view, pending conflicts must be addressed by the TC at the time of integration.
If any additional action is needed from our side, please, let us know.

**Comment by carlospzurita** [comment]
We followed you're recommendations to merge the `next` branch on this one and applied `./gradlew spotlessApply`.  We now see some quality isses from the Codacy bot that affect old code. How shall we proceed with this PR?

File: src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java [hunk 1]
```diff
@@ -217,12 +217,33 @@ public void init() throws ParseException, ConfigurationException, IOException, S
...
         final TimedExpiredItemsRemover timedExpiredItemsRemover = new TimedExpiredItemsRemover();
-        timedExpiredItemsRemover.addExpirationItemHolder(
-                (l, timeUnit) -> taskPoolRegistry.removeDone(),
-                0, TimeUnit.HOURS);
+        timedExpiredItemsRemover.addExpirationItemHolder((l, timeUnit) -> taskPoolRegistry.removeDone(), 0,
+                TimeUnit.HOURS);
         // 7,5 minutes
         timer.scheduleAtFixedRate(timedExpiredItemsRemover, 450000, 450000);
 
+        String maxThreads = etfConfig.getProperty("etf.testruns.threads.max");
```
Comment (jonherrmann) [inline src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java]: The configuration logic must be done in the EtfConfigController

File: src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java [hunk 2]
```diff
@@ -105,10 +105,11 @@
 
     private final static String TEST_RUNS_URL = API_BASE_URL + "/TestRuns";
 
-    public final static int MAX_PARALLEL_RUNS = Runtime.getRuntime().availableProcessors();
+    public static int MAX_PARALLEL_RUNS;
```
Comment (jonherrmann) [inline src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java]: do not use this style for none _final static_ variables

File: src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java [hunk 3]
```diff
@@ -217,12 +217,33 @@ public void init() throws ParseException, ConfigurationException, IOException, S
...
+            } else {
+                throw new ConfigurationException(maxThreads + " is not a valid value for etf.testruns.max.threads");
+            }
+        }
+        String maxQueues = etfConfig.getProperty("etf.testruns.queued.max");
+        try {
+            MAX_QUEUE_SIZE = Integer.parseInt(maxQueues);
+        } catch (NumberFormatException e) {
+            if ("auto".equals(maxQueues)) {
+                MAX_QUEUE_SIZE = Runtime.getRuntime().availableProcessors() * 3;
```
Comment (jonherrmann) [inline src/main/java/de/interactive_instruments/etf/webapp/controller/TestRunController.java]: = maxParallelRuns * 3

**Comment by carlospzurita** [comment]
We have pushed the requested changes on this pull request. Some issues have appeared from the codacy report, but we think that they can be marked as solved, as we have followed the same code style and practices that were present on the original code. However, if you have any feedback, or see fit to make more changes, please let us know.

**Pull request merged**